index.html

```diff
@@ -650,33 +650,52 @@ <h3 id="requesting-more-permission">
         </h3>
         <p>
           To <dfn data-lt="request permission to use|requesting permission to use" class=
-          "export">request permission to use</dfn> a |descriptor:PermissionDescriptor|, the user
-          agent must perform the following steps. This algorithm returns either
-          {{PermissionState/"granted"}} or {{PermissionState/"denied"}}.
+          "export">request permission to use</dfn> a |descriptor:PermissionDescriptor|, with
+          optionally a |promise:Promise|:
```

Comment:
We need to make call sites easy to read, and I don't think "request permission to use |descriptor| with |promise|" explains to readers what the promise is going to mean. Readers shouldn't need to click through in order to read the call site, which means the call site needs to mention [=transient activation=]. I think something like
```suggestion
          "export">request permission to use</dfn> a |descriptor:PermissionDescriptor|, optionally
          with [=transient activation=] consumed through a |promise:Promise|:
```
would be somewhat better, although "consumed through" is still obscure, and you might come up with better ideas.

Reply:
In that case you should define a named parameter: https://infra.spec.whatwg.org/#algorithm-params. (It's not clear to me how a caller would invoke the suggestion.)